bundles/org.openhab.binding.cloudrain/src/main/java/org/openhab/binding/cloudrain/internal/handler/CloudrainZoneHandler.java

```diff
@@ -0,0 +1,427 @@
+
+    /**
+     * A reference to a {@link TimeZoneProvider} for localized times
+     */
+    private TimeZoneProvider timeZoneProvider;
+
+    /**
+     * The update job which provides real time status updates for active irrigations
+     */
+    private @Nullable ScheduledFuture<?> remainingSecsUpdateJob;
+
+    /**
+     * A reference to the {@link CloudrainAPI}
+     */
+    private CloudrainAPI cloudrainAPI;
+
+    /**
+     * The id if the zone associated with this handler
+     */
+    private String zoneId = "";
+
+    /**
+     * a local attribute for convenient access to the defined command duration. Default value 10 seconds.
+     */
+    private transient int localCommandDuration = 10;
+
+    /**
+     * This class handles real time status update of active irrigations.
+     * If scheduled each second this job will decrease the remaining seconds accordingly.
+     * If configured the job will also update the zone's status at the projected end of the irrigation.
+     */
+    private class RealtimeUpdateJob implements Runnable {
+        private int remaining;
+        private Channel channel;
+        private CloudrainConfig config;
+
+        public RealtimeUpdateJob(Channel channel, int remaining, CloudrainConfig config) {
+            this.channel = channel;
+            this.remaining = remaining;
+            this.config = config;
+        }
+
+        @Override
+        public void run() {
+            if (--remaining >= 0) {
+                if (config.getRealtimeUpdates()) {
+                    updateState(channel.getUID(), new DecimalType(remaining));
+                }
+            } else {
+                // trigger thing update to reflect the end of the irrigation if desired
+                if (config.getUpdateAfterIrrigation()) {
+                    updateStatusFromAPI();
+                }
+            }
+        }
+    }
+
+    /**
+     * Creates this CloudrainZoneHandler
+     *
+     * @param thing the Cloudrain Zone thing
+     * @param cloudrainAPI the initialized Cloudrain API. Handlers of this class expect that the
+     *            {@link CloudrainAccountHandler} has already authenticated.
+     * @param timeZoneProvider the TimeZoneProvider for handling dates
+     * @param itemChannelLinkRegistry required for checking this Zone's channel link status
+     * @param timeZoneProvider a {@link TimeZoneProvider} for date time localization
+     */
+    public CloudrainZoneHandler(Thing thing, CloudrainAPI cloudrainAPI, TimeZoneProvider timeZoneProvider,
+            ItemChannelLinkRegistry itemChannelLinkRegistry) {
+        super(thing);
+        this.cloudrainAPI = cloudrainAPI;
+        this.timeZoneProvider = timeZoneProvider;
+        this.itemChannelLinkRegistry = itemChannelLinkRegistry;
+    }
+
+    @Override
+    public void initialize() {
+        // Retrieve this zone's ID from the properties as set by discovery
+        String zoneIdFromProperty = getThing().getProperties().get(PROPERTY_ZONE_ID);
```

Comment:
This should be retrieved via the Thing's config to allow textual configuration, too.